homeassistant/components/ring/sensor.py

```diff
@@ -69,18 +69,33 @@
     sensors = []
     for device in ring_chimes:
         for sensor_type in SENSOR_TYPES:
-            if "chime" in SENSOR_TYPES[sensor_type][1]:
-                sensors.append(RingSensor(hass, device, sensor_type))
+            if "chime" not in SENSOR_TYPES[sensor_type][1]:
+                continue
+
+            if sensor_type in ("wifi_signal_category", "wifi_signal_strength"):
+                await hass.async_add_executor_job(device.update)
```

Comment:
Why don't we call the hub update function in the component before we forward the entry so that all devices have their data in order before the platforms are set up?

Reply:
https://github.com/home-assistant/home-assistant/blob/f50714d7e99abe771a1cb634c01d5cb7950f861b/homeassistant/components/ring/__init__.py#L119

Reply:
because it makes a ton of requests and it's overkill. The whole Ring lib needs to clean up request handling. Each device hits a shared info endpoint and extracts just their info from it.

Reply:
(rewriting that whole lib is for another day)

Reply:
Well this is what's currently happening every scan update interval anyway. Doing it on startup too won't make it that much worse.